client-src/default/socket.js

```diff
@@ -4,11 +4,21 @@ const SockJS = require('sockjs-client/dist/sockjs');
 
 let retries = 0;
 let sock = null;
+let messageQueue = [];
 
-const socket = function initSocket(url, handlers) {
+function send(message) {
+  if (sock.readyState === SockJS.OPEN) return sock.send(message);
+  messageQueue.push(message);
+}
+
+const socket = function initSocket(url, handlers, messages) {
   sock = new SockJS(url);
 
   sock.onopen = function onopen() {
+    messageQueue.forEach(function dequeue(message) {
+      sock.send(message);
+    });
+    messageQueue = [];
```

Comment:
Breaking change for multi compiler compilation

Reply:
Never remove messages, because you can have two/three/four clients, and some clients can have delay

Reply:
How do you mean? If you lose connection and reconnect you probably don't want to resend all messages again?

Reply:
I've added the message queue for messages that are sent before the connection is fully established or else you will get an error from sockjs.

Reply:
@EloB open two or three browsers and tests how it works, we already do same before and remove this, because it is break situation when you have more then one client (some client can disconnect and connect again)

Reply:
That `initSocket` looks really weird in the first place because it's not pure and does side effects on `sock`.

Reply:
Maybe we could rewrite that `initSocket` to be pure instead?